api/src/registry.rs

```diff
@@ -191,6 +220,18 @@ impl ImageConfig {
             .filter_map(|port| u16::from_str(port.as_str()).ok())
             .min()
     }
+
+    fn attached_volume(&self) -> Option<Vec<String>> {
```

Comment:
I would go with following

```suggestion
    fn attached_volume(&self) -> Vec<&String> {
```

- There should be no need for cloning…
- An empty vec or a none vec same same space in memory: https://users.rust-lang.org/t/space-time-usage-to-construct-vec-t-vs-option-vec-t/35596
- And I agree with:
  > There is no point in wrapping a collection into an Option just for the sake of it. It only makes code more complex, and introduces additional conditional jumps when you need to access the contents.

Reply:
Clone is removed, Vec<&String> is returned as suggested